fe/fe-core/src/test/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvRewriteOptimizationTest.java

```diff
@@ -324,6 +346,394 @@ public void testNestedMvOnSingleTable() throws Exception {
+                " group by v1, test_all_type.t1b");
+        String query6 = "SELECT t0.v1 as v1, " +
+                " sum(test_all_type.t1c) as total_sum, count(test_all_type.t1c) as total_num" +
+                " from t0 join test_all_type" +
+                " on t0.v1 = test_all_type.t1d" +
+                " where t0.v1 < 99" +
+                " group by v1";
+        // rollup test
+        String plan6 = getFragmentPlan(query6);
+        PlanTestBase.assertContains(plan6, "1:AGGREGATE (update finalize)\n" +
+                "  |  output: sum(18: total_sum), sum(19: total_num)\n" +
+                "  |  group by: 16: v1\n" +
+                "  |  \n" +
+                "  0:OlapScanNode\n" +
+                "     TABLE: agg_join_mv_2\n" +
+                "     PREAGGREGATION: ON\n" +
+                "     PREDICATES: 16: v1 <= 98");
+        dropMv("test", "agg_join_mv_2");
+
+        createAndRefreshMv("test", "agg_join_mv_3", "create materialized view agg_join_mv_3" +
+                " distributed by hash(v1) as SELECT t0.v1 as v1," +
+                " test_all_type.t1b, sum(test_all_type.t1c) * 2 as total_sum," +
+                " count(distinct test_all_type.t1c) + 1 as total_num" +
+                " from t0 join test_all_type on t0.v1 = test_all_type.t1d" +
+                " where t0.v1 < 100" +
+                " group by v1, test_all_type.t1b");
+
+        // rollup with distinct
+        String query7 = "SELECT t0.v1 as v1, " +
+                " (sum(test_all_type.t1c)  * 2) + (count(distinct test_all_type.t1c) + 1) as total_sum," +
+                " (count(distinct test_all_type.t1c) + 1) * 2 as total_num" +
+                " from t0 join test_all_type" +
+                " on t0.v1 = test_all_type.t1d" +
+                " where t0.v1 < 99" +
+                " group by v1";
+        String plan7 = getFragmentPlan(query7);
+        PlanTestBase.assertNotContains(plan7, "agg_join_mv_3");
+
+        // distinct rewrite without rollup
+        String query8 = "SELECT t0.v1, test_all_type.t1b," +
+                " (sum(test_all_type.t1c) * 2) + 1 as total_sum, (count(distinct test_all_type.t1c) + 1) * 2 as total_num" +
+                " from t0 join test_all_type" +
+                " on t0.v1 = test_all_type.t1d" +
+                " where t0.v1 < 99" +
+                " group by v1, test_all_type.t1b";
+        String plan8 = getFragmentPlan(query8);
+        PlanTestBase.assertContains(plan8, "agg_join_mv_3");
+
+        dropMv("test", "agg_join_mv_3");
+    }
+
+    @Test
+    public void testUnionRewrite() throws Exception {
+        // single table union
+        createAndRefreshMv("test", "union_mv_1", "create materialized view union_mv_1" +
+                " distributed by hash(empid)  as select empid, deptno, name, salary from emps where empid < 3");
+        String query1 = "select empid, deptno, name, salary from emps where empid < 5";
+        getFragmentPlan(query1);
```

Comment:
No Check?

Reply:
I add a check for single table rewrite because it is determinant